addons/account/models/mail_message.py

```diff
@@ -167,16 +204,17 @@ def _except_audit_log(self):
             # Invoicing/Customers menu)
             has_related_move = self.env['account.move'].sudo().search_count([
                 ('partner_id', 'in', partner_message.account_audit_log_partner_id.ids),
-                ('company_id.check_account_audit_trail', '=', True),
             ], limit=1)
             if not has_related_move:
                 to_check -= partner_message
         for message in to_check:
-            if message.account_audit_log_activated and not (
-                message.account_audit_log_move_id
-                and not message.account_audit_log_move_id.posted_before
-            ):
-                raise UserError(self.env._("You cannot remove parts of the audit trail."))
+            if message.account_audit_log_move_id and message.account_audit_log_move_id.state == 'draft':
+                return
```

Comment:
You can't return like this in a loop
```suggestion
                continue
```